tests/functional/t/typevar_name_missing_variance.py

```diff
@@ -0,0 +1,35 @@
+"""Test case for typevar-name-missing-variance."""
+
+from typing import TypeVar
+
+# Type variables without variance
+T = TypeVar("T")
+T_co = TypeVar("T_co")  # [typevar-name-missing-variance]
+T_contra = TypeVar("T_contra")  # [typevar-name-missing-variance]
+
+# Type variables not starting with T_
+N = TypeVar("N")
+N_co = TypeVar("N_co", covariant=True)
```

Comment:
I really need to read the documentation about that, my review is not helpful right now. I trust Marc's opinion if you want to merge this in 2.12.